.github/workflows/prod-deploy.yml

```diff
@@ -1,67 +1,136 @@
+                per_page: 1,
+              }
+            )
+
+            const [issue] = searchResult.items
+            if (!issue) {
+              throw new Error('No release PR found')
+            }
+
+            // Getting PR data from /pulls API has more info about the pull
+            // request like merge_commit_sha.
+            const { data: pr } = await github.rest.pulls.get({
+              owner,
+              repo,
+              pull_number: issue.number,
+            })
+
+            return pr
 
   tag-frontend-release:
     concurrency:
       group: tag-frontend-release-${{ github.ref }}
       cancel-in-progress: true
 
     outputs:
-      tag: frontend-v${{ steps.extract-version.outputs.result }}
+      tag: frontend-v${{ steps.get-tag-name.outputs.result }}
 
     name: tag frontend release
     runs-on: ubuntu-latest
-    if: ${{ (github.repository == 'chanzuckerberg/cryoet-data-portal') && (needs.create-frontend-release.outputs.prs_created == 'true') }}
-    needs: create-frontend-release
+    if: github.repository == 'chanzuckerberg/cryoet-data-portal'
+    needs: find-frontend-release-pr
     steps:
       - name: merge release PR
         uses: actions/github-script@v7
         id: merge-release-pr
+        if: fromJson(needs.find-frontend-release-pr.outputs.pr).state == 'open'
         with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
           result-encoding: string
           script: |
-            const pr = ${{ needs.create-frontend-release.outputs.pr }}
+            const pr = ${{ needs.find-frontend-release-pr.outputs.pr }}
             console.log(`Merging release PR #${pr.number}`)
 
-            const [owner, repo] = '${{ github.repository }}'.split('/')
+            const { owner, repo } = context.repo
+            const mergeResponse = await github.rest.pulls.merge({
+              owner,
+              repo,
+              pull_number: pr.number,
+              merge_method: 'squash',
+            })
 
-            const [mergeResponse] = await Promise.all([
-              github.rest.pulls.merge({
-                owner,
-                repo,
-                pull_number: pr.number,
-                merge_method: 'squash',
-              }),
+            return mergeResponse.data.sha
+
+      - name: checkout repo for tagging
+        uses: actions/checkout@v4
+        with:
+          ref: |
+            ${{
+              fromJson(needs.find-frontend-release-pr.outputs.pr).state == 'open' &&
+              steps.merge-release-pr.outputs.result ||
+              fromJson(needs.find-frontend-release-pr.outputs.pr).merge_commit_sha
+            }}
+
+      - name: get tag name
+        id: get-tag-name
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const pr = ${{ needs.find-frontend-release-pr.outputs.pr }}
+            const version = pr.title.split(' ').at(-1)
+
+            return {
+              version,
```

Comment:
Are we using the version from this anywhere? If not, would it be cleaner just to have it return the tag, so we don't have to parse the json wherever we are trying to get the tag?